### PR TITLE
GUVNOR-2431: Redesign Copy, Save, Rename, Restore and Delete Popups

### PR DESCRIPTION
**Pull request by manstis**
Fix use of PathProvider.

(Plus change to lamda's)

@karreiro Would you mind reviewing please?

### CODE REVIEW

File: uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderImpl.java [hunk 1]
```diff
@@ -263,57 +238,54 @@ public BasicFileMenuBuilder addCopy( final Command command ) {
...
+                                                       getCopyPopupCommand( copyCaller,
+                                                                            path,
+                                                                            copyPopUpPresenter.getView() ) ) );
     }
 
     public BasicFileMenuBuilder addCopy( final PathProvider provider,
                                          final Validator validator,
                                          final Caller<? extends SupportsCopy> copyCaller ) {
-        return addCopy( provider.getPath(), validator, copyCaller );
+        return addCopy( () -> {
```
Comment (manstis) [inline uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderImpl.java]: This is the guts of the fix. Move call to ```provider.getPath()``` from "menu construction" to "menu execution"

File: uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderTest.java [hunk 2]
```diff
@@ -104,12 +104,16 @@ public void setup() {
...
     public void testDelete() {
-        builder.addDelete( provider, deleteCaller );
+        builder.addDelete( provider,
+                           deleteCaller );
 
         final Menus menus = builder.build();
         final MenuItem mi = menus.getItems().get( 0 );
         final MenuItemCommand mic = (MenuItemCommand) mi;
 
+        verify( provider,
```
Comment (manstis) [inline uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderTest.java]: Should not call ```provider.getPath()``` until the menu item's ```Command``` is executed.

File: uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderTest.java [hunk 3]
```diff
@@ -125,6 +129,9 @@ public void testRename() {
...
```
Comment (manstis) [inline uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderTest.java]: Should not call ```provider.getPath()``` until the menu item's ```Command``` is executed.

File: uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderTest.java [hunk 4]
```diff
@@ -141,6 +148,9 @@ public void testCopy() {
...
```
Comment (manstis) [inline uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderTest.java]: Should not call ```provider.getPath()``` until the menu item's ```Command``` is executed.

**Comment by manstis** [comment]
Jenkins please retest this.

File: uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderTest.java (hunk 2 above)
Comment (karreiro) [inline uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/menu/BasicFileMenuBuilderTest.java]: Nice test coverage :)

**Comment by karreiro** [comment]
+ 1 to merge. Thanks @manstis

**Comment by ederign** [comment]
thanks @manstis @karreiro

**Pull request merged**